### PR TITLE
feat(python): Add DataFrame.glimpse()

### PR DESCRIPTION
**Pull request by zundertj**
See #5550.

### CODE REVIEW

File: py-polars/polars/internals/dataframe/frame.py [hunk 1]
```diff
@@ -2468,10 +2468,84 @@ def filter(
...
+        max_col_name = max((len(col_name) for col_name, _, _ in data))
+
+        # dtype string
+        max_col_dtype = max((len(dtype_str) for _, dtype_str, _ in data))
+
+        # limit the amount of data printed such that total width is fixed
+        max_col_values = 100 - max_col_name - max_col_dtype
+
+        # print header
+        print(f"Rows: {self.height}")
```
Comment (ritchie46) [inline py-polars/polars/internals/dataframe/frame.py:2530]: I think we should construct a `str` and return that. That way the user can decide what to do with it later. Send it to `stderr`, `stdout`, a logging utility, etc.

Comment (zundertj) [reply py-polars/polars/internals/dataframe/frame.py]: Done

**Comment by zundertj** [comment]
I don't understand why the newline is taken as a literal on Linux but as a newline on Windows here..

**Comment by ritchie46** [comment]
Hmm... Maybe we should skip the doctest check?

**Comment by zundertj** [comment]
Not very satisfying to ignore the result, but better than not having the feature I guess..

**Comment by ritchie46** [comment]
And we are green! :) Thanks a lot @zundertj

**Pull request merged**